### PR TITLE
Implement complete recurring events support

### PR DESCRIPTION
**Pull request by mpnordland**
Helping to complete issue #173, adds the --until (or -u) option to specify an until date. It also adds the beginnings of editing recurring events in ikhal. Currently, just frequency is editable.

### CODE REVIEW

File: khal/aux.py [hunk 1]
```diff
@@ -72,6 +72,9 @@ def datetimefstr(date_list, datetimeformat, longdatetimeformat,
     :returns: a datetime
     :rtype: datetime.datetime
     """
+    print("longdatetimeformat:", longdatetimeformat)
+    print("datetimeformat:", datetimeformat)
+    print(date_list)
```
Comment (hobarrera) [inline khal/aux.py]: Are you sure you meant to leave these `print` lines?

**Comment by hobarrera** [comment]
Can you add test cases, at least for the cli usage?

**Comment by mpnordland** [comment]
Whoops, I'll get those removed. I'll also get some tests added.

**Comment by hobarrera** [comment]
It also looks like one of the current tests if failing with the changes, so double check that if you can. :)

Thanks!

**Comment by geier** [comment]
please also add yourself to `AUTHORS.txt`

**Comment by hobarrera** [comment]
Looks like there are some style violations, according to [travis](https://travis-ci.org/geier/khal/jobs/61686260). I'd suggest you set up your editor (IDE?) to run flake8 on your code automatically if you can - though you can just run it manually if you prefer, it's just good habit. :) .

It would seem that running `tox` manually doesn't do the style checking (though travis is configured to run it). This most likely never came up because all the developers have had some form of style checking locally.
Maybe we should address this at some point.

**Comment by geier** [comment]
Thanks @mpnordland, I just merged this into master.

**Comment by untitaker** [comment]
The tox-issue should be resolved by adding the `style` testenv to the envlist in tox.ini.

**Pull request merged**